[yidgha] initial submission of yidgha

mnjames opened the pull request.

mnjames commented:
The checks have passed. Could you let me know when this will be available for download from Keyman?

mcdurdin commented on release/y/yidgha/README.md:
```diff
@@ -0,0 +1,35 @@
+Description
+-----------
+
+This keyboard layout is designed for Balochi. It includes 
```
Is this the correct language name?

mcdurdin commented on release/y/yidgha/source/yidgha.js:
```diff
@@ -0,0 +1 @@
+KeymanWeb.KR(new Keyboard_yidgha());function Keyboard_yidgha(){this.KI="Keyboard_yidgha";this.KN="Yidgha Keyboard";this.KMINVER="9.0";this.KV={F:' 1em "Arial"',K102:0};this.KDU=0;this.KV.KLS={"default": ["","١","٢","٣","٤","٥","٦","٧","٨","٩","٠","ـ","","","","","ق","و","ع","ر","ت","ے","ݮ","ى","ه","پ","","","","","","","ا","س","د","ف","گ","ح","ج","ك","ل","څ","ݯ","","","","","","","ز","ش","ث","ط", ...
```
This file does not need to be in the repo. It's an output file.

mcdurdin commented on release/y/yidgha/source/yidgha.kps:
```diff
@@ -0,0 +1,72 @@
+  </System>
+  <Options>
+    <ExecuteProgram></ExecuteProgram>
+    <MSIFileName>D:\Documents\Keyman Developer 10.0\yidgha\keymandesktop-10.0.1078.0\keymandesktop100.msi</MSIFileName>
```
The `MSIFileName` element should be empty

mcdurdin commented on release/y/yidgha/source/yidgha.kmn:
```diff
@@ -0,0 +1,84 @@
+﻿store(&VERSION) '9.0'
+store(&NAME) 'Yidgha Keyboard'
```
Best not to use the word "Keyboard" in the name of the keyboard

rachitralli commented:
@mcdurdin I have already submitted pull request for Yidgha please see 

https://github.com/rachitralli/Rachitrali-Yidgha-Keyboard 

this is the correct and final keyboard for yidgha language.

rachitralli commented:
@mcdurdin @LornaSIL @DavidLRowe @MakaraSok I have already submitted pull request for Yidgha keyboard in yidgha language spoken in Pakistan please see

https://github.com/rachitralli/Rachitrali-Yidgha-Keyboard

this is the correct and final keyboard for yidgha language. please check, Thanks

mcdurdin commented:
@rachitralli Please note, this is a different and unrelated Yidgha keyboard.

mnjames commented:
I don't understand how to change the .kvk file into a .kvks file. I have Deveoper 10 but it seems to be saving the file as a .kvk file.

Also, should I delete that from my /sources directory since it seems to show up in the /build directory when I build the keyboard?

mnjames commented:
@rachitralli If you're interested in working with the Forum for Language Initiatives (FLI) I'm sure they would be happy to collaborate with you. You could possibly merge keyboards or agree between yourselves which keyboard to keep. I'm simply helping FLI post this to Keyman and am not personally involved in the development.
I will point out that your keyboards do not appear in the Keyman download options from the Android app (and likely from other Keyman programs). As a result no one at FLI knew about the existence of your keyboards.

LornaSIL commented:
* Rename ydg.ico to yidgha.ico to match folder name (then change .kpj, .kps and .kmn to reference new name) -- actually, the ydg.ico seems to be buggy. You may need to recreate it as yidgh.ico. 
* In .kps change ```<Language ID="ydg">ydg</Language>``` to ```<Language ID="ydg">Yidgha</Language>```
* I would prefer that you use the Scheherazade font already in the repo. If you do that, you'll need to
     * Remove Scheherazade-Regular.ttf from the PR
     * Change the path in the .kpj to the font in ```release\shared\fonts\sil\scheherazade``` The path in the file will probably look like this: ```<Filepath>source\..\..\..\shared\fonts\sil\scheherazade\Scheherazade-Regular.ttf</Filepath>``` (this font may not have been in the shared folder at the time you did this PR as we only added it a week ago).
* We need a readme.htm and a welcome.htm which are then called in the .kps and .kpj. 
     * A readme.htm is displayed before installing the keyboard. It can be as simple as saying what language the keyboard is for (the same information you have in README.md)
     * A welcome.htm describes how to use the keyboard. It can be as simple as having a screenshot or more complicated (html/css). In Keyman Developer go to your on-screen keyboard and click on "Export". Then choose between html or image (png). Most of us use the image so that there isn't a problem with displaying the font (as an example, sil_areare in the welcome.htm uses a screenshot). ```O``` and ```P``` outputs are not showing up on the keyboard. You may want to document that in writing rather than as a graphic since U+061C is not a visible character.
* .kmn works fine
* The touch layout works correctly. I do have a few comments:
     * If the touch and phone layout are the same, I would recommend removing one of them. That way you don't have to keep two keyboards in sync (I only tested the tablet layout).
     * In the touch layout there are keys that aren't displaying anything on the keycap, but they produce something. I'd either delete them so they don't take up real estate or add something to them. They are: K_BKQUOTE, K_EQUAL, K_LBRKT, K_RBRKT, K_BKSLASH, K_oE2, K_PERIOD. There are even more of them on the shift state.
* We've had people decide to add "Latin-style digits" to their keyboards later on (especially on the mobile). You might consider whether you should go ahead and add them now or not.

Let me know if I haven't been clear. I know I've put a lot of feedback in here, but hopefully it will be clear what needs changing. It's really in a good state.

rachitralli commented:
@mnjames thanks for your message. I have developed 68 keyboards for the languages of Pakistan, Kashmir,  Afghanistan and India and now donating these keyboards to keyman via github free of cost. I am a linguist and volunteer from Pakistan and not charging a single penny from keyman for these keyboards. I have developed these keyboards in Developer 6 and now converting these keyboards to Developer 9 and 10, I am also developing touch versions for these keyboards, some are available for download for  computer as well as touch devices. cheer, thanks

mcdurdin commented:
@mnjames in terms of the .kvk vs .kvks situation, you should rename the .kvk in the source/ folder to a .kvks, and then Keyman Developer will take care of the rest if you load the keyboard source and save it (may have to do a spurious edit to get Keyman Developer to treat the keyboard as modified for save).

You do need to have the .kvks (or .kvk if you haven't renamed it) in the source folder. The duplicate in the build/ folder is generated from the source/ copy (either by direct copy or by compile, depending on whether you are using .kvk or .kvks respectively).

mcdurdin commented:
Sorry, forgot to mention that after renaming the .kvk to .kvks, you'll need to update the `store(&visualkeyboard)` reference to the .kvks as well, probably in a text editor, before attempting to load the keyboard in Keyman Developer.

mnjames commented:
Here's the current state:

- I've changed the .kvk file into a .kvks file.
- I've created a new icon file. I don't actually know how to do this and so watched a YouTube video which showed me how to create an icon with multiple sizes. If that doesn't work I can change it. Also, I couldn't find any documentation on what size icon Keyman expects.
- .kps language name corrected.
- Scheherazade font changed to the one in the respository.
- I created readme.htm and welcome.htm files. I added the welcome.htm file into my .kps package but not the readme.htm file (because that's what someone else did in Balochi). I don't really understand what things do or don't need to be in that package.
- I deleted the tablet touch keyboard and left just the phone touch keyboard. I think that's what @LornaSIL was referring to but I'm not positive.
- The undefined keys in the touch layout (and also the on screen keyboard) are also undefined in the main layout. I'm a little confused about whether if a key is undefined it won't work (what I would expect) or whether it will type the character from the default keyboard on whatever device you're using (what someone else told me happens).
  - For example, if [K_LBRKT] is left undefined, will it actually type a [ symbol?
- I won't add Latin numbers at this time. We expect to modify the keyboard in the future based on user feedback. Right now I'm just trying to figure out how to get keyboards up on github and into the apps.

rachitralli commented:
@mnjames please add touch layout if possible

LornaSIL commented:
This is almost ready.
* Add the readme.htm to both the .kps and .kpj files
     * After adding it in the .kps, go to the "Details" tab and select it as the "Readme file". That's optional, but it does pop up when the user installs the package which is nice.
* Add the graphics for the welcome.htm file to both the .kpj and .kps files
* Remove the .kmn from the .kps file (unless you want the source distributed in the .kmp)
I still need to test the mobile layout so I'll add more if I find anything else.

LornaSIL commented:
You have a rule in the keyboard for "M" to output a space. Is that what you want?
```
+ [SHIFT K_M] > ' '
```
Regarding the touch layout, the blank keys are outputting something. If they are going to output something, I'd like there to be something on the keycap to show the expected behaviour.
* Default layer
     * K_EQUAL > put  = on the keycap
     * K_LBRKT > put [ on the keycap
     * K_RBRKT > put ] on the keycap
     * K_BKSLASH > put \ on the keycap
     * K_BKQUOTE> put ` on the keycap
     * K_PERIOD> put . on the keycap
* Shift layer
     * K_1 .. K_8 > put  !@#$%^&* on the keycaps
     * K_P > I don't have any good ideas on how to show ALM, maybe put an "A"? "ALM" won't display, it has too many characters.
     * K_M > This is outputting a space because of the rule in the keyboard. If you don't want that, delete the rule.
     * K_LBRKT > put { on the keycap
     * K_RBRKT > put } on the keycap
     * K_COLON > put : on the keycap
     * K_QUOTE > put " on the keycap
     * K_BKSLASH > put | on the keycap
     * K_COMMA > put < on the keycap
     * K_PERIOD> put > on the keycap
     * K_BKQUOTE> put ` on the keycap
     * K_SLASH> put ? on the keycap

mnjames commented:
- readme.html and welcome.htm images added to .kps
- removed .kmn from .kps
  - also moved .kmx file inside .kps to point to build/ folder. I'm not sure if I'm supposed to share the build folder, but it doesn't seem like compiled files should be included in the source/ folder.
- removed [SHIFT K_M] entry
- added "auto-fill underlying layout" to On-Screen and Touch Layouts. I think this will solve the list Lorna made but I'm not sure.

rachitralli commented:
@LornaSIL you are right help files are very necessary. thanks

LornaSIL commented:
@mnjames
https://keyman.com/keyboards/yidgha
